### PR TITLE
feat(sentry): added annotation to specify organization per compoment

### LINKED ISSUE #11688: [Sentry] Allow to overwrite organization for components
The Sentry plugin allows passing only an organization inside the backstage configuration file. But this is not too good if I want to use backstage for different Sentry organizations.

## Feature Suggestion

I suggest adding the possibility to overwrite the organization in a component using a new annotation:
```yaml
sentry.io/organization: anotherOrg
```

WDYT?

### PR DESCRIPTION
**Pull request by antoniomuso**
## Hey, I just made a Pull Request!

I added the possibility to add an annotation to specify the sentry organization!

close #11688 
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/sentry/src/api/annotations.ts [hunk 1]
```diff
@@ -17,7 +17,12 @@
 import { Entity } from '@backstage/catalog-model';
 
 export const SENTRY_PROJECT_SLUG_ANNOTATION = 'sentry.io/project-slug';
+export const SENTRY_ORGANIZATION_ANNOTATION = 'sentry.io/organization';
```
Comment (freben) [inline plugins/sentry/src/api/annotations.ts:20]: Can't we bake this into the other annotation? So you can specify `my-org/my-project`, but in a backwards compatible way such that things still work if it doesn't have that first part.

Would update the docs here then https://github.com/backstage/backstage/blob/179a5906560a8d5b2cc4ff3b229dd76a91387eca/docs/features/software-catalog/well-known-annotations.md?plain=1#L249

**Review by freben — COMMENTED** [review]
Thanks!

File: plugins/sentry/src/api/production-api.ts [hunk 2]
```diff
@@ -26,10 +28,18 @@ export class ProductionSentryApi implements SentryApi {
...
 
   async fetchIssues(
-    project: string,
-    statsFor: string,
-    query?: string,
+    entity: Entity,
+    {
+      query,
+      statsFor,
+    }: {
```
[the remark this answers is not in the document]
Comment (antoniomuso) [reply plugins/sentry/src/api/production-api.ts]: Done!

File: plugins/sentry/src/api/annotations.ts (hunk 1 above)
Comment (antoniomuso) [reply plugins/sentry/src/api/annotations.ts]: Done!

File: .changeset/bright-buckets-deliver.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-sentry': minor
+---
+
+Added annotation to specify organization per component
```
Comment (Rugvip) [inline .changeset/bright-buckets-deliver.md:5]: Could you add a `**BREAKING**: ...` section as well that mentions the API change? Don't think there's a need to go into details though, just the fact that it changed.

**Review by Rugvip — CHANGES_REQUESTED** [review]
Nice! 👍 

Just a tiny detail

File: .changeset/bright-buckets-deliver.md (hunk 3 above)
Comment (antoniomuso) [reply .changeset/bright-buckets-deliver.md]: Done!

File: .changeset/purple-cherries-fix.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+---
+'@backstage/plugin-sentry': major
+---
+
+Added the possibility to specify organization per component
+
+**BREAKING**: Now the annotation `sentry.io/project-slug` can have the format of `[organization]/[project-slug]` or just `[project-slug]`.
```
Comment (Rugvip) [inline .changeset/purple-cherries-fix.md:7]: That change is actually not breaking, but tbh I think it's a better description so perhaps add it to the line above?

The breaking change is that change to the `fetchIssues` method signature

**Review by Rugvip — CHANGES_REQUESTED** [review]

File: .changeset/purple-cherries-fix.md (hunk 4 above)
Comment (antoniomuso) [reply .changeset/purple-cherries-fix.md]: Done!

File: .changeset/purple-cherries-fix.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+'@backstage/plugin-sentry': major
+---
+
+Added the possibility to specify organization per component, now the annotation `sentry.io/project-slug` can have the format of `[organization]/[project-slug]` or just `[project-slug]`.
+
+**BREAKING**: The method `fetchIssue` changed the signature:
+
+```diff
+export interface SentryApi {
+  // (undocumented)
```
Comment (Rugvip) [inline .changeset/purple-cherries-fix.md:11]:
```suggestion
```

File: .changeset/purple-cherries-fix.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
+---
+'@backstage/plugin-sentry': major
```
Comment (Rugvip) [inline .changeset/purple-cherries-fix.md:2]:
```suggestion
'@backstage/plugin-sentry': minor
```

Just a minor bump for now, don't wanna move to 1.x yet

**Review by Rugvip — CHANGES_REQUESTED** [review]
Tiny tweaks. Don't worry about the CI failures, I think that's some trouble with docker that'll probably sort itself out on the rerun

**Pull request merged**